extension/goplugin/schemas.go

```diff
@@ -281,26 +297,45 @@ func (schema *Schema) fetchImpl(id string, requestContext goext.Context, fetch f
 	return schema.ResourceFromMap(data)
 }
 
-// Fetch fetches a resource by id.
-// Schema, Logger, Environment and pointer to raw resource are required fields in the resource
-func (schema *Schema) Fetch(id string, context goext.Context) (interface{}, error) {
-	fetched, err := schema.FetchRaw(id, context)
+type rawFetchFunc func() (interface{}, error)
+
+func (schema *Schema) fetch(fetch rawFetchFunc) (interface{}, error) {
+	fetched, err := fetch()
 	if err != nil {
 		return nil, err
 	}
 	xRaw := reflect.ValueOf(fetched)
 	return schema.rawToResource(xRaw)
 }
 
+// Fetch fetches a resource by id.
+// Schema, Logger, Environment and pointer to raw resource are required fields in the resource
+func (schema *Schema) Fetch(id string, context goext.Context) (interface{}, error) {
+	return schema.fetch(func() (interface{}, error) {
+		return schema.FetchRaw(id, context)
```

Comment:
calling a public function from another public function looks suspicious. why don't we extract the common fetch implementation and parametrize it with conver/no convert and lock/no lock policy?